pkg/companion-api/backend/client.go

```diff
@@ -7,6 +7,7 @@ import (
 // UserData is a simple user struct with paswordhash and claims
 type UserData struct {
 	PasswordHash string           `json:"password"`
+	OauthToken   string           `json:"oauth_token"`
```

Comment:
I'd expected multiple oauth identities bound to a single account here (ie: 1 google, 1 github, etc)

Reply:
ok. It means that for each client account (ie: 1 google, 1 github, etc.), 1 companion-api will have to be deployed. Or it is also necessary that the companion-api can manage the "n-account" with several CLIENTID, ... etc?

Reply:
I'd say we want something like `<prefix>/oauth/<provider>/<client-id>` pointing to the associated user account.

Reply:
what I mainly meant was to use this path as an *etcd* storage key, ie associating user `foo` with clientid `bar` in provider `baz` is `kv.Put("oauth/baz/bar", "foo")` (and the symmetric `Get` to check a clientid knowning the source provider to get the user id to query).

Reply:
but the multiprovider support is also required and LGTM ;)